feat(curriculum): add sum all numbers algorithm (#60409)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lab-sum-all-numbers-algorithm/6827cc540b8818ebf41d581d.md

---
id: 6827cc540b8818ebf41d581d
title: Design a Sum All Numbers Algorithm
challengeType: 26
dashedName: lab-sum-all-numbers-algorithm
---

# --description--

In this lab, you will need to design a sum all numbers algorithm.

> 

Reviewer: ```suggestion

Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

```

> Your `sumAll` function accepts an array of two numbers. Your function should return the sum of those two numbers plus the sum of all the numbers between them. The lowest number will not always come first.
> 
> For example, `sumAll([4,1])` should return `10` because sum of all the numbers between 1 and 4 (both inclusive) is `10`.

Reviewer: ```suggestion
1. You should have a function named `sumAll` function that accepts an array of two numbers.
1. `sumAll([n, m])` should return the sum `n` and `m` plus the sum of all the numbers between them. The lowest number will not always come first. For example, `sumAll([4,1])` should return `10` because sum of all the numbers between `1` and `4` (both inclusive) is `10`.
```

# --hints--

> 

Reviewer: ````suggestion

You should have a function named `sumAll`.

```js
assert.isFunction(sumAll);
```

````

`sumAll([1, 4])` should return a number.

```js
assert.isNumber(sumAll([1, 4]));
```

`sumAll([1, 4])` should return `10`.

```js
assert.equal(sumAll([1, 4]), 10);
```

`sumAll([4, 1])` should return `10`.

```js
assert.equal(sumAll([4, 1]), 10);
```

`sumAll([5, 10])` should return `45`.

```js
assert.equal(sumAll([5, 10]), 45);
```

`sumAll([10, 5])` should return `45`.

```js
assert.equal(sumAll([10, 5]), 45);
```

# --seed--

## --seed-contents--

```js

> function sumAll(arr) {
> 
> }
> 
> sumAll([1, 4]);

Reviewer: ```suggestion

```
